pennylane/templates/embeddings.py

```diff
@@ -0,0 +1,207 @@
+(represented by the gate parameters) into a quantum state (see also Schuld & Killoran 2019
+:cite:`schuld2019`).
+
+Provided embeddings
+--------------------
+
+Qubit architectures
+*******************
+
+.. autosummary::
+
+    AngleEmbedding
+    AmplitudeEmbedding
+    BasisEmbedding
+
+Continuous-variable architectures
+*********************************
+
+.. autosummary::
+
+    SqueezingEmbedding
+    DisplacementEmbedding
+
+Code details
+^^^^^^^^^^^^
+"""
+#pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane import RX, RY, RZ, BasisState, Squeezing, Displacement, QubitStateVector
+
+
+def AngleEmbedding(features, n_wires, rotation='X'):
+    r"""
+    Uses the entries of ``features`` as rotation angles of qubits.
+
+    The details of the strategy are defined by the `rotation` parameter:
+     * `rotation = 'X'` uses the features to chronologically apply Pauli-X rotations to qubits
+     * `rotation = 'Y'` uses the features to chronologically apply Pauli-Y rotations to qubits
+     * `rotation = 'Z'` uses the features to chronologically apply Pauli-Z rotations to qubits
+
+    The length of ``features`` has to be smaller or equal to the number of qubits. If there are fewer entries in ``features``
+    than rotations, the circuit does not apply the remaining rotation gates.
+
+    This embedding method can also be used to encode a binary sequence into a basis state. Choose rotation='X'
+    and features of a nonzero value of :math:`\pi /2` only where a qubit has to be prepared in state 1.
+
+    Args:
+        features (array): Input array of shape ``(N, )``, where N is the number of input features to embed
+        n_wires (int): Number of qubits in the circuit
+        rotation (str): Strategy of embedding
+
+    """
+
+    if len(features) > n_wires:
+        raise ValueError("Number of features to embed cannot be larger than num_wires, "
+                         "but is {}.".format(len(features)))
+    if rotation == 'X':
+        for op in range(len(features)):
+            RX(features[op], wires=op)
+
+    elif rotation == 'Y':
+        for op in range(len(features)):
+            RY(features[op], wires=op)
+
+    elif rotation == 'Z':
+        for op in range(len(features)):
+            RZ(features[op], wires=op)
+
+    else:
+        raise ValueError("Rotation has to be `X`, `Y` or `Z`, got {}.".format(rotation))
+
+
+def BasisEmbedding(basis_state, n_qubits):
+    r"""Prepares a quantum state in the state ``basis_state``.
+
+    For example, for ``basis_state=[0, 1, 0]``, the quantum system will be prepared in state :math:`|010 \rangle`.
+
+    .. note::
+
+        BasisEmbedding uses PennyLane's :func:`BasisState()` and only works in conjunction with
+        devices that implement this function.
+
+    Args:
+        features (array): Input array of shape ``(N, )``, where N is the number of input features to embed
+        n_qubits (int): Number of qubits in the circuit
```

Comment:
Should we use `n_wires` here for consistency?

Reply:
Done